Assets/MixedRealityToolkit.Providers/ObjectMeshObserver/RoomFileFormat/RoomFileSerializer.cs

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SpatialObjectMeshObserver.RoomFile
+{
+    /// <summary>
+    /// Converts a UnityEngine.Mesh object to and from an array of bytes that conform to the .room file format.
+    ///    File header: vertex count (32 bit integer), triangle count (32 bit integer)
+    ///    Vertex list: vertex.x, vertex.y, vertex.z (all 32 bit float)
+    ///    Triangle index list: 32 bit integers
+    /// </summary>
+    public static class RoomFileSerializer
+    {
+        /// <summary>
+        /// The mesh header consists of two 32 bit integers.
+        /// </summary>
+        private static int HeaderSize = sizeof(int) * 2;
+
+        /// <summary>
+        /// Serializes a list of Mesh objects to the specified writer.
+        /// Optionally transforms the vertices into the supplied secondarySpace.
+        /// </summary>
+        /// <param name="writer">The writer to which to write the serialized objects.</param>
+        /// <param name="meshes">List of MeshFilter objects to be serialized.</param>
+        /// <param name="secondarySpace">New space to transform the vertices into.</param>
+        public static void Serialize(BinaryWriter writer, IList<MeshFilter> meshes, Transform secondarySpace = null)
```

Comment:
Do we support / want to support serializing to the HTK's format? Or is this left over and unneeded now that we support additional "real" formats?

Reply:
it's not really needed anymore. i can remove if you think it is the right thing to do

Reply:
Serialize and Write* removed as they are not needed for the asset importer and end up being stale code.